aya/src/programs/xdp.rs

```diff
@@ -174,7 +176,9 @@ impl Xdp {
     ///
     /// Ownership of the link will transfer to this program.
     pub fn attach_to_link(&mut self, link: XdpLink) -> Result<XdpLinkId, ProgramError> {
-        let prog_fd = self.data.fd_or_err()?;
+        let prog_fd = self.fd()?;
```

Comment:
Here and everywhere else, why not do:

```
let prog_fd = self.fd()?.as_fd().as_raw_fd();
```

Reply:
Just personal preference - I prefer to avoid `?` in the middle of an expression. This will also make the diff cleaner when we get around to removing the need for `as_raw_fd`; it'll be a pure deletion.